test/epoch_token_locker.js

```diff
@@ -0,0 +1,223 @@
+      await epochTokenLocker.increaseStateIndex()
+      await epochTokenLocker.updateDepositsBalanceTest(user_1, ERC20.address)
+
+      assert.equal( (await epochTokenLocker.getBalance.call(user_1, ERC20.address)).toNumber(), 100)
+    })
+    it("returns just the balance + pending deposit, if there are no withdraws", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+      
+      await epochTokenLocker.deposit(ERC20.address, 100)
+      await epochTokenLocker.increaseStateIndex()
+      assert.equal(await epochTokenLocker.getBalance.call(user_1, ERC20.address), 100)
+    })
+    it("returns just the balance + pending deposit - depending withdraws", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+      
+      await epochTokenLocker.deposit(ERC20.address, 100)
+      await epochTokenLocker.requestWithdraw(ERC20.address, 50)
+      await epochTokenLocker.increaseStateIndex()
+      assert.equal(await epochTokenLocker.getBalance.call(user_1, ERC20.address), 50)
+    })
+    it("returns just the balance + pending deposit - depending withdraws and protects overflows", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+      
+      await epochTokenLocker.deposit(ERC20.address, 100)
+      await epochTokenLocker.requestWithdraw(ERC20.address, 150)
+      await epochTokenLocker.increaseStateIndex()
+      assert.equal(await epochTokenLocker.getBalance.call(user_1, ERC20.address), 0)
+    })
+    it("returns just the balance + pending deposit if withdraw was made in same stateIndex", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
+      await ERC20.givenAnyReturnBool(true)
+      
+      await epochTokenLocker.deposit(ERC20.address, 100)
+      await epochTokenLocker.increaseStateIndex()
+      await epochTokenLocker.requestWithdraw(ERC20.address, 150)
+      assert.equal(await epochTokenLocker.getBalance.call(user_1, ERC20.address), 100)
+    })
+  })
+  describe("addBalance", () => {  
+    it("modifies the balance by adding", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
+
+      await epochTokenLocker.addBalanceTest(user_1, ERC20.address, 100)
+
+      assert.equal(await epochTokenLocker.getBalance(user_1, ERC20.address), 100)
+    })
+  })
+  describe("substractBalance", () => {  
+    it("modifies the balance by substracting", async () => {
+      const epochTokenLocker = await EpochTokenLockerTestInterface.new()
+      const ERC20 = await MockContract.new()
```

Comment:
Interesting! Should be be doing this instead of actually using `MintableERC20` in the Snapp contract tests? Or is this just an "easy-way-out"